superset-frontend/src/dashboard/containers/DashboardPage.tsx

```diff
@@ -237,6 +237,15 @@ const DashboardPage: FC = () => {
     return () => {};
   }, [css]);
 
+  useEffect(
+    () => () => {
+      // need to clean up custom css
+      // when dashboard unmounts
+      injectCustomCss('');
```

Comment:
Looking at line 235, it seems like the function `injectCustomCss` was designed to handle the cleanup (or that was the intention).
I would recommend reviewing that function and finding out why is it not working.
The `useEffect` from above should be sufficient then.